velox/dwio/parquet/writer/Writer.h

```diff
@@ -94,6 +95,7 @@ struct WriterOptions {
   // The default factory allows the writer to construct the default flush
   // policy with the configs in its ctor.
   std::function<std::unique_ptr<DefaultFlushPolicy>()> flushPolicyFactory;
+  std::shared_ptr<arrow::Schema> schema;
```

Comment:
@JkSelf we should use `std::shared_ptr<const Type> schema` in the Velox public api.
We can then probably create an empty vector from `schema` and use `exportToArrow` to get the arrow schema.

Reply:
@majetideepak Updated. I am not familiar with macos build. Can you help to look at the build failed issue in macos env? Thanks.